hosting_docs/source/installation/2-manual-install.rst

```diff
@@ -175,6 +175,45 @@ our example cluster, the control machine is named “control1”.
 
         $ sudo -H pip install ansible virtualenv virtualenvwrapper --ignore-installed six
 
+Install Python 3.10 (optional)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Python 3.10 requires additional setup on Ubuntu 18.04 to work properly.
+
+1.  Install system packages
+
+    ::
+
+        $ sudo add-apt-repository -y ppa:deadsnakes/ppa
+        $ sudo apt update
+        $ sudo apt-get -y install python3.10 python3.10-dev python3.10-distutils libffi-dev
+
+2.  Install updated version of pip for use with python 3.10
+
+    ::
+
+        $ sudo python3 -m pip install --target=/usr/local/lib/python3.10/dist-packages pip setuptools
```

Comment:
Is this a documented and supported way to do that? I'd be curious to see the source if it's documented somewhere.

The typical way to get pip on Ubuntu is by installing the apt package `python3-pip`. Does deadsnakes not provide a `python3.10-pip` package?

Reply:
Not documented that I know of. This is our alternative approach to the steps documented in https://github.com/pypa/setuptools/issues/2988#issuecomment-1007991935 which is messier than this approach. I could split this out to only install `pip` with the special `--target` syntax, and then install `setuptools` the usual way since pip works at that point.

Reply:
> I could split this out to only install `pip`...

If we are forced to use this technique then yes, the less we install this way the better. It seems fragile to me, and I wouldn't be surprised if it doesn't always work.

Reiterating earlier question: Does deadsnakes not provide a `python3.10-pip` package?

Reply:
Offlined with Daniel and while there is not a pip package available via deadsnakes, if we install `python3.10-venv`, and create a virtual environment with 3.10, pip works. Going to update the directions to reflect this as it is a safer approach.

Reply:
The syntax to create virtual environments is slightly different between venv and virtualenv. When thinking through how to roll this change out, it like creating a separate changelog for switching from virtualenv to venv would be necessary.

However upon further inspection, `venv` is available without needing to install python3.10-venv. Additionally, it appears available on our machines running 3.6 that do not have python3-venv or python3.6-venv installed, which means we could shift to venv syntax without breaking existing workflows. Daniel did mention that he initially had issues using `python3 -m venv <path>` to create virtualenvs, which is why `virtualenv` was added to begin with. I am not observing that now, but we should confirm this behavior before moving forward.